### ISSUE TITLE
Tweak: Don't inherit parent tag when creating child slot through the scene inspector

### ISSUE THREAD

**Issue by JackTheFoxOtter** [Yellow-Dog-Man/Resonite-Issues#1379]
### Is your feature request related to a problem? Please describe.

Currently, when creating a child slot on a slot that has a tag in the inspector, the tag of the parent is copied over to the newly created child. We are currently working on a system identifying certain slots in our world through tags, and this default behaviour has caused issues multiple times already, when unintentionally tagging the children as something they shouldn't be tagged as without realizing.

### Describe the solution you'd like

I'd like to tweak the default behavior so that the tag of the paren't isn't copied over when creating a child slot on it.

### Describe alternatives you've considered

Manually removing the tag after creating the child. It's not the end of the world, but it has been the cause of confusion multiple times in recent days, and I believe setting leaving the tag of the child empty would make more sense as the default behavior.

### Additional Context

I assume there was a reason to make this the default behavior back in the day. Therefore I assume there are use-cases in which this behavior does make sense over just leaving the tag empty. Personally, I cannot think of a good reason to justify leaving the default behavior as it is, so if you know good reasons for it, please note them down here! I'm curious.

**Comment by stiefeljackal** [comment]
Based on experimenting with this behavior on a previous FrooxEngine version, the reason why this default behavior of tagging exists is due to mainly the permission system. With some permissions components such as `GrabbablePermissions` and `TouchablePermissions`, you can define a list of tags that you can either Blacklist or Whitelist for particular roles. The reason why created slots inherit the tag from the parent is to retain the integrity of the permission.

As an example, both these components (`GrabbablePermissions` and `TouchablePermissions`) have the tag **Developer** under the Blacklist when creating a default world. When spawning an Inspector, it will be tagged with **Developer**, and anything that is created within that inspector panel (the tree view and the slot information) will be given the same tag, hence prevent users without the appropriate role from interacting with the panel. You can can inspect the slots on any inspector and see the **Developer** tag on all of them.**

<img width="1193" alt="image" src="https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/20023996/ad52cafa-dd52-4cbe-acf9-a55c74d6fdbb">

This tagging behavior is not limited to creating empty slots under a tagged slot. If you have not tried already, place a user under a slot with a tag on it. Primitives created, model imports, and ProtoFlux nodes spawned will inherit the same tag as well. The only actions, at least to my knowledge, that do not inherit the tag is duplicating slots and spawning items.

I hope this gives you more insight on this behavior. Having to clear out a tag after creating a child slot is rather tedious when you are still in the building and development phase, so at least having the option of not inheriting parent tags automatically might be beneficial.

**Comment by JackTheFoxOtter** [comment]
Interesting. So this is basically an implicit "security" feature of the engine, rather than an explicit behavior you have to specify with a component. Personally I'm not super confident this makes sense at least in the case of manually adding a slot through the inspector though. If you get to the point where you can create a child slot through the inspector, you will also be able to change the tag, making this very ineffective as a security mechanism.

What happens when you spawn an item that already has a tag on the root slot while parented under a tagged slot yourself? Would that overwrite the tag? Because if so, this might actually be somewhat concerning and could break functionality that expects a spawned item to have a certain tag depending on how the world is set up. (I'll test this myself later, just can't access Resonite right now for a couple of hours).

Personally I'd prefer the auto-tagging to be something explicit instead, so something like a AutoTagNewChildren-Component. This would also make it more useful, as you could specify a different tag for the children that doesn't have to match the parent tag exactly.

**Comment by stiefeljackal** [comment]
> If you get to the point where you can create a child slot through the inspector, you will also be able to change the tag, making this very ineffective as a permission / security mechanism.

In a default world, that would require the **Builder** or **Admin** role to interact with the Inspector since the **Moderator**, **Guest**, and **Spectator** roles cannot interact with the Inspector due to the Inspector (particularly the Create Child button) being tagged as Developer.

<img width="469" alt="image" src="https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/20023996/820882cb-0b9d-4c42-9321-9e716da806d1">

With that said, I am unsure how this would be ineffective as a permission mechanism if only the **Admin** or **Builder** can create child slots through the Inspector. If these two roles decide to change a tag on the created slot or change the tags on the Inspectors that will cause negative side effects, then it is on them.

### Moderator and Guest Touchable Permissions

<img width="454" alt="image" src="https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/20023996/6e3813c0-6e33-470a-8bfc-9e00435a955e">

### Spectator Touchable Permissions

<img width="695" alt="image" src="https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/20023996/665b05c5-b58c-431c-b0dc-d42ddacedfeb">

> What happens when you spawn an item that already has a tag on the root slot while parented under a tagged slot yourself? Would that overwrite the tag? Because if so, this might actually be somewhat concerning and could break functionality that expects a spawned item to have a certain tag depending on how the world is set up. (I'll test this myself later, just can't access Resonite right now for a couple of hours).

As I stated in my first post, duplicating slots or spawning items _do not_ inherit the tag of the parent slot.

**Comment by JackTheFoxOtter** [comment]
> In a default world, that would require the Builder or Admin role to interact with the Inspector since the Moderator, Guest, and Spectator roles cannot interact with the Inspector due to the Inspector (particularly the Create Child button) being tagged as Developer.

Exactly. So slots created through the inspector wouldn't have to inherit the tag, as when you can click that button, you can also change the tag. When you can't change the tag, you also can't click that button.

> As I stated in my first post, duplicating slots or spawning items do not inherit the tag of the parent slot.

You are correct, and that makes a lot of sense. Apologies for misreading!
